Subject: Pppdump: Fixed building with GCC 15 (#548)
Origin: upstream, commit v2.5.2-4-g0536169 <https://github.com/ppp-project/ppp/commit/0536169>
Author: Jaroslav Škarvada <jskarvad@redhat.com>
Date: Thu Feb 27 23:00:16 2025 +0100
Bug-Debian: https://bugs.debian.org/1097645

    GCC 15 defaults to C23 which does not allow K&R declarations.

    Credit Yaakov Selkowitz in:
    https://src.fedoraproject.org/rpms/ppp/pull-request/12

    Signed-off-by: Jaroslav Škarvada <jskarvad@redhat.com>

--- a/pppdump/pppdump.c
+++ b/pppdump/pppdump.c
@@ -42,14 +42,12 @@
 extern int optind;
 extern char *optarg;
 
-void dumplog();
-void dumpppp();
-void show_time();
+void dumplog(FILE *);
+void dumpppp(FILE *);
+void show_time(FILE *, int);
 
 int
-main(ac, av)
-    int ac;
-    char **av;
+main(int ac, char **av)
 {
     int i;
     char *p;
@@ -97,8 +95,7 @@
 }
 
 void
-dumplog(f)
-    FILE *f;
+dumplog(FILE *f)
 {
     int c, n, k, col;
     int nb, c2;
@@ -241,8 +238,7 @@
 unsigned char dbuf[8192];
 
 void
-dumpppp(f)
-    FILE *f;
+dumpppp(FILE *f)
 {
     int c, n, k;
     int nb, nl, dn, proto, rv;
@@ -375,9 +371,7 @@
 }
 
 void
-show_time(f, c)
-    FILE *f;
-    int c;
+show_time(FILE *f, int c)
 {
     time_t t;
     int n;
